### PR TITLE
Properly populate `tags` in `kwargs` for `Potentials`

### LINKED ISSUE #600: Tests are Failing with gmso 0.7.0
**Bug summary**

Since the release of gmso 0.7.0, tests are failing because of a bug where gmso's `AbstractPotential` is not saving tags for the given atom types. 

**Code to reproduce the behavior**
An example of this failure is given below:
https://github.com/mosdef-hub/foyer/blob/5679d7068e99832b93b96c306a110d9377359339/foyer/tests/test_gmso_forcefield.py#L233-L256

This fails because we rely on `atom_type.tags` to get element information:
https://github.com/mosdef-hub/foyer/blob/5679d7068e99832b93b96c306a110d9377359339/foyer/general_forcefield.py#L156-L164

And the tags are empty because of a refactor for the metadata: **Curplit : https://github.com/mosdef-hub/gmso/pull/589**

Since this originated in foyer, I am opening the issue here, but will be transferred to gmso.

### PR DESCRIPTION
**Pull request by umesh-timalsina**
See #600, For a summary of the bug. This PR fixes #600.

### CODE REVIEW

File: gmso/tests/test_atom_type.py [hunk 1]
```diff
@@ -16,6 +16,12 @@ class TestAtomType(BaseTest):
     def atomtype_metadata(self):
         return AtomType()
 
+    def test_atom_type_tag_kwarg(self):
+        at = AtomType(
+            tags={"element": "Li", "comesFrom": "ForceFieldExperiments"}
+        )
+        assert at.tag_names == ["element", "comesFrom"]
+
```
Comment (justinGilmer) [inline gmso/tests/test_atom_type.py:24]: Should we also add a test that ensures that the tags, if empty, return an empty list?

**Review by justinGilmer — APPROVED** [review]
Small comment, but LGTM

**Review by daico007 — APPROVED** [review]
LGTM!
